### PR TITLE
Correct updates to TestItem.error and TestItem.sortText (Fixes #166190)

### LINKED ISSUE #166190: TestItem.error and TestItem.sortText updates not shown correctly in Testing panel
Does this issue occur when all extensions are disabled?: N/a

- VS Code Version: 
- OS Version: Version: 1.74.0-insider (user setup)
Commit: ef415578d1b1019db585718968a303eeca4b8979
Date: 2022-11-11T05:19:14.666Z
Electron: 19.1.3
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Windows_NT x64 10.0.19044
Sandboxed: Yes

Steps to Reproduce:

1. Build this sample extension (test-provider-sample) - https://github.com/r3m0t/vscode-extension-samples/tree/test-sort-error-fix
2. Press 'refresh' in Testing panel repeatedly. Expected: the 'error' message should update and the 'Order' children should switch order. Actual: the children get updated labels, but still rendered in the old order. And the 'error' message doesn't update.

Fix incoming

### PR DESCRIPTION
**Pull request by r3m0t**
See #166190 for test plan

### CODE REVIEW

File: src/vs/workbench/contrib/testing/browser/explorerProjections/hierarchalByLocation.ts [hunk 1]
```diff
@@ -153,9 +153,9 @@ export class HierarchicalByLocationProjection extends Disposable implements ITes
 						break;
 					}
 
-					// parent needs to be re-rendered on an expand update, so that its
+					// parent needs to be re-rendered on an expand update or its children have changed order, so that its
 					// children are rewritten.
-					const needsParentUpdate = existing.test.expand === TestItemExpandState.NotExpandable && patch.expand;
+					const needsParentUpdate = (existing.test.expand === TestItemExpandState.NotExpandable && patch.expand) || patch.item?.sortText;
```
Comment (connor4312) [inline src/vs/workbench/contrib/testing/browser/explorerProjections/hierarchalByLocation.ts:158]: I think the fix for error/uri is okay, but changes to the sortText should ideally result in calls to `tree.resort()` rather rerendering the parent.

Comment (r3m0t) [reply src/vs/workbench/contrib/testing/browser/explorerProjections/hierarchalByLocation.ts]: Not sure about the uri change actually, as it's a readonly field. I am working on the resort change- specifically improving the tests to also test the TreeSorter implementation.

**Comment by r3m0t** [comment]
@connor4312 yep, all done

**Comment by connor4312** [comment]
Thanks for the PR!

**Pull request merged**